### PR TITLE
compiler.h changes for Windows MSVC support

### PR DESCRIPTION
**Pull request by tarun292**
Differential Revision: D65328573

### CODE REVIEW

**Review by dbort — APPROVED** [review]

File: kernels/prim_ops/register_prim_ops.cpp [hunk 1]
```diff
@@ -148,7 +148,7 @@ static Kernel prim_ops[] = {
           EValue& out = *stack[2];
           if (a.isInt() && b.isInt()) {
             const int64_t quot = a.toInt() / b.toInt();
-            if (std::signbit(a.toInt()) == std::signbit(b.toInt())) {
+            if ((a.toInt() < 0) == (b.toInt() < 0)) {
```
Comment (malfet) [inline kernels/prim_ops/register_prim_ops.cpp:151]: This feels unrelated to PR in question, is it?

File: runtime/platform/compiler.h [hunk 2]
```diff
@@ -100,22 +100,38 @@
 #endif // (__cplusplus) >= 202002L
 
 /// Define a C symbol with weak linkage.
+#ifdef _MSC_VER
+// There currently doesn't seem to be a great way to do this in Windows and
+// given that weak linkage is not really critical on Windows, we'll just leave
+// it as a stub.
+#define ET_WEAK
```
Comment (chinazhangchao) [inline runtime/platform/compiler.h:107]: Hi, we are adding `Windows x64 Build Support` https://github.com/pytorch/executorch/pull/6979. I use `.\install_requirements.bat --pybind xnnpack` to build on windows. And we encountered duplicate symbol problem. 
image
I investigated and found that windows didn't support weak symbol. Do you have some insights how to resolve this?
